### PR TITLE
Add backup matching

### PR DESCRIPTION
**Pull request by SeanDuHare**
**Description**
Currently some SldyImages fail to read due to some inconsistent read behavior with `fsspec.glob`. This seems to happen specifically with the `LocalFileSystem` so I added a backup matching system using `os`. 

**Testing**
I tested this with a file sent to me that was failing and was able to read it successfully.

### CODE REVIEW

File: bioio_sldy/sldy_image.py [hunk 1]
```diff
@@ -259,6 +257,30 @@ def metadata(self) -> typing.Dict[str, typing.Optional[dict]]:
...
+            glob_matcher = self.image_directory / f"{self._data_file_prefix}*.npy"
+            data_path_matches = self._fs.glob(f"{glob_matcher}")
+            self._data_paths = set(
+                [pathlib.Path(data_path) for data_path in data_path_matches]
+            )
+
+            if not self._data_paths:
+                self._data_paths = {
+                    self.image_directory / filename
+                    for filename in os.listdir(f"{self.image_directory}/")
```
Comment (toloudis) [inline bioio_sldy/sldy_image.py]: is this trailing slash always ok?

**Review by toloudis — APPROVED** [review]
Looks good, thank you for the fix!

File: bioio_sldy/sldy_image.py [hunk 2]
```diff
@@ -259,6 +257,30 @@ def metadata(self) -> typing.Dict[str, typing.Optional[dict]]:
 
         return self._metadata
 
+    @property
+    def data_paths(self) -> typing.Set[pathlib.Path]:
+        if not self._data_paths:
+            glob_matcher = self.image_directory / f"{self._data_file_prefix}*.npy"
+            data_path_matches = self._fs.glob(f"{glob_matcher}")
```
Comment (toloudis) [inline bioio_sldy/sldy_image.py:264]: Is there a known fsspec issue about this?

File: bioio_sldy/sldy_image.py (hunk 1 above)
Comment (SeanDuHare) [reply bioio_sldy/sldy_image.py]: Nope! I meant to use pathlib for this, nice catch

File: bioio_sldy/sldy_image.py (hunk 2 above)
Comment (SeanDuHare) [reply bioio_sldy/sldy_image.py:264]: They had an issue I referenced in the issue that limits this repos `fsspec` that reportedly fixed it, but it did not seem to for us during my testing.

**Review by BrianWhitneyAI — APPROVED** [review]

**Pull request merged**